### PR TITLE
582-bug-fix-baseline-dnsmap

### PR DESCRIPTION
**Pull request by Cyriacjose**
Using Rendercomponent instead of user guide
Adding inline and other important comments
Aligning Code for better presentation and removing some structure and grammar issues

### CODE REVIEW

File: src/components/DNSMap/DNSMap.tsx [hunk 1]
```diff
@@ -3,25 +3,14 @@ import { useForm } from "@mantine/form";
 import { useCallback, useState } from "react";
 import { CommandHelper } from "../../utils/CommandHelper";
 import ConsoleWrapper from "../ConsoleWrapper/ConsoleWrapper";
-import { UserGuide } from "../UserGuide/UserGuide";
+import { RenderComponent } from "../UserGuide/UserGuide";
 import { SaveOutputToTextFile_v2 } from "../SaveOutputToFile/SaveOutputToTextFile";
 import { LoadingOverlayAndCancelButton } from "../OverlayAndCancelButton/OverlayAndCancelButton";
 import { l } from "vitest/dist/index-60e2a8e1";
```
Comment (michaeljpigott) [inline src/components/DNSMap/DNSMap.tsx:10]: You need to import checkAllCommandsAvailability. See Airbase-NG for reference.

Comment (michaeljpigott) [inline src/components/DNSMap/DNSMap.tsx:10]: You need to import InstallationModal. See Airbase-NG for reference.

File: src/components/DNSMap/DNSMap.tsx [hunk 2]
```diff
@@ -30,14 +19,39 @@ interface FormValuesType {
...
+    //Componenets state Variable
+    const [loading, setLoading] = useState(false); // State variable that represents loading state of the component
+    const [output, setOutput] = useState(""); // State variable that represents the output text from the DNS mapping process
+    const [checkedAdvanced, setCheckedAdvanced] = useState(false); //State variable that represents the state of advanced mode switch
+    const [Pid, setPid] = useState(""); //State variable that represents the process ID of the DNS mapping process
+    const [allowSave, setAllowSave] = useState(false); //State variable that represents whether saving the output is allowed
+    const [hasSaved, setHasSaved] = useState(false); //State variable that represents whether the output has been saved
+
+    //Components Constant Variables
+    const title = "DNS Mapping for Subdomains (DNSMap)";
```
Comment (michaeljpigott) [inline src/components/DNSMap/DNSMap.tsx:37]: Need to match the title with what's on Kali.org, so it should be Dnsmap. 
```suggestion
    const title = "Dnsmap";
```

File: src/components/DNSMap/DNSMap.tsx [hunk 3]
```diff
@@ -30,14 +19,39 @@ interface FormValuesType {
...
+        "Step 1: Enter a valid domain to be mapped.\n" +
+        "       Eg: google.com\n\n" +
+        "Step 2: Enter a delay between requests. Default is 10 (milliseconds). Can be left blank.\n" +
+        "       Eg: 10\n\n" +
+        "Step 3: Click 'Start Mapping' to commence the DNSMap tool's operation.\n\n" +
+        "Step 4: View the Output block below to view the results of the tool's execution.\n\n" +
+        "Switch to Advanced Mode for further options.";
+    const sourceLink = "https://www.kali.org/tools/dnsmap/"; // Link to the source code (or Kali Tools).
+    const tutorial = ""; // Link to the official documentation/tutorial.
+    const dependencies = ["DNSMap"]; // Contains the dependencies required by the component.
```
Comment (michaeljpigott) [inline src/components/DNSMap/DNSMap.tsx:52]: Should be all lowercase. Needs to match the command entered into the terminal. 
```suggestion
    const dependencies = ["dnsmap"]; // Contains the dependencies required by the component.
```

File: src/components/DNSMap/DNSMap.tsx [hunk 4]
```diff
@@ -134,41 +148,51 @@ const DNSMap = () => {
...
-            </Stack>
-        </form>
+        <RenderComponent
+            title={title}
+            description={description_userguide}
+            steps={steps}
+            tutorial={tutorial}
+            sourceLink={sourceLink}
+        >
+            <form onSubmit={form.onSubmit(onSubmit)}>
```
Comment (michaeljpigott) [inline src/components/DNSMap/DNSMap.tsx:158]: Need to add the loadingModal here. Refer to Airbase-ng.

**Review by michaeljpigott — CHANGES_REQUESTED** [review]
Hi @Cyriacjose 

This is looking good. You need to add the checkAllCommandsAvailability and InstallationModal components so DDT checks to see if dnsmap has been installed. Refer to Airbase-ng for reference. 

Kind regards

Michael

**Comment by michaeljpigott** [comment]
@Cyriacjose Can you also please add the issue that you are working on to this pull request?

**Comment by Cyriacjose** [comment]
Fix for Issue: Bug Fix (Baseline): DNSMap

**Review by michaeljpigott — CHANGES_REQUESTED** [review]
Hi @Cyriacjose 

Good work. You just need to implement the checkAllCommandsAvailability in your code. Refer to airbase-ng to see how to do it. 

Kind regards

Michael

**Review by michaeljpigott — APPROVED** [review]

**Pull request merged**